In [1]:
import numpy as np

# Задание 1

Напишите уравнение параболы, проходящей через три точки (x,y): `(1,2)`, `(3,10)`, `(5,1)`.

Парабола задается квадратичной функцией: $$y=ax^{2}+bx+c \quad при \quad a\neq0$$

Подставив данные точки в выражение, получим СЛАУ:

$$\left\{\begin{matrix}
a+b+c=2\\ 
9a+3b+c=10\\ 
25a+5b+c=1
\end{matrix}\right.$$

### Решение с помощью обратной матрицы:

In [2]:
A = np.array([1,1,1,9,3,1,25,5,1]).reshape((3,3))
print('Матрица A:')
print(A)

Матрица A:
[[ 1  1  1]
 [ 9  3  1]
 [25  5  1]]


In [3]:
B = np.array([2,10,1]).reshape((3,1))
print('Матрица B:')
print(B)

Матрица B:
[[ 2]
 [10]
 [ 1]]


In [4]:
np.linalg.det(A)

-16.000000000000007

In [5]:
inv_A = np.linalg.inv(A)
print('Матрица, обратная A:')
print(inv_A)

Матрица, обратная A:
[[ 0.125 -0.25   0.125]
 [-1.     1.5   -0.5  ]
 [ 1.875 -1.25   0.375]]


In [6]:
X = np.dot(inv_A,B)
X

array([[-2.125],
       [12.5  ],
       [-8.375]])

In [7]:
print('a = ', X[0])
print('b = ', X[1])
print('c = ', X[2])

a =  [-2.125]
b =  [12.5]
c =  [-8.375]


Проверка с помощью метода `linalg.solve`

In [8]:
np.linalg.solve(A,B)

array([[-2.125],
       [12.5  ],
       [-8.375]])

### Ответ:
Уравнение параболы:
$$y=-2\frac{1}{8}x^{2}+12\frac{1}{2}x-8\frac{3}{8}$$

# Задание 2
Известно, что свежий огурец на 99% состоит из воды. Месяц назад взвесили мешок со свежими огурцами. Получилось, что огурцов ровно 100 кг. Мешок убрали, а через месяц снова взвесили. Огурцы за это время усохли, и теперь вода составляет уже только 98% их веса. Сколько теперь (в кг) весят огурцы?

### Решение через пропорции:
Мешок 100 кг огурцов примем за 100% массы. Тогда `x` кг воды составляет 99% массы.

$$\left\{\begin{matrix}
100\rightarrow 100\%\\ 
x\rightarrow 99\%
\end{matrix}\right.$$

$$x=\frac{100\cdot99}{100}=99$$

Таким образом, масса воды в свежих огурцах составляет 99 кг.

Масса безводного вещества в огурцах постоянна, её можно выразить как $(100-x)$ кг.

$$(100-99)=1$$

Таким образом, масса безводного вещества в огурцах составляет 1 кг.

После усыхания `у` кг огурцов примем за 100% массы. Масса воды в огурцах будет равняться разности массы усохших огурцов и массы безводного вещества в огурцах и будет составлять 98% от всей массы.

$$\left\{\begin{matrix}
y\rightarrow 100\%\\ 
(y-1)\rightarrow 98\%
\end{matrix}\right.$$

$$100(y-1)=98y$$

$$2y=100$$

$$y=50$$
### Ответ:
Таким образом, после усыхания огурцы имеют массу **50 кг**.

# Задание 3. 
### Определение логарифма. Решить уравнения:

* 1) 8.0
* 2) 8.228818690495881
* 3) 2
* 4) 4
* 5) x1=1/9, x2=3

# Задание 4. 
### Свойства логарифмов. Вычислить:


* 6) 2.0
* 7) -2.0
* 8) 0.5
* 9) 1.5
* 10) 2.0
* 11) 2.0
* 12) 5
* 13) 2.0
* 14) 1.5
* 15) 5